crates/client/rpc/src/lib.rs

```diff
@@ -517,6 +517,43 @@ where
         })
     }
 
+    /// Estimate the fee associated with transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - starknet transaction request
+    /// * `block_id` - hash of the requested block, number (height), or tag
+    ///
+    /// # Returns
+    ///
+    /// * `fee_estimate` - fee estimate in gwei
+    async fn estimate_fee(&self, request: BroadcastedTransaction, block_id: BlockId) -> RpcResult<FeeEstimate> {
+        // TODO:
+        //      - modify BroadcastedTransaction to assert versions == "0x100000000000000000000000000000001"
```

Comment:
Do you plan to merge with this TODO ? Is there anything blocking to implement it ? If yes can we at least create an issue and reference it here ?